pkg/services/dashboards/database/database.go

```diff
@@ -688,6 +688,14 @@ func (d *dashboardStore) deleteChildrenDashboardAssociations(sess *db.Session, d
 
 	if len(dashIds) > 0 {
 		for _, dash := range dashIds {
+			if checkProvisioning {
+				// #TODO do a batch check for provisioning data
+				// Allow GetProvisionedDataByDashboardID to accept a slice or do a direct sql query
+				if provisioningData, _ := d.GetProvisionedDataByDashboardID(context.Background(), dash.Id); provisioningData != nil {
```

Comment:
So when you delete a folder, it looks like it will run a query for every dashboard in the folder. This could cause pretty significant performance issues if there are a lot of dashboards.

I think an easy fix would be to do this check in a single query when the user tries to delete a folder.

Reply:
Thanks for the review! I had added it as a todo above the line you commented on so I'll make sure to implement that if there are no objections to the overall approach.

Reply:
lol I'm blind. Sorry, didn't even see that